### PR TITLE
Protection against using the wrong sample alias which produces zero L…

### PR DESCRIPTION
**Pull request by yfarjoun**
…OD. Now the tool will fail (i.e. return 1)

### Description

_Give your PR a **concise** yet **descriptive** title_
_Please explain the changes you made here._
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
_You can delete these instructions once you have written your PR description._

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/fingerprint/CheckFingerprint.java [hunk 1]
```diff
@@ -365,11 +366,22 @@ protected int doWork() {
...
+            if (metrics.LOD_EXPECTED_SAMPLE!=0) {
+                anyNonzeroLod |= true;
+            }
         }
 
         summaryFile.write(outputSummaryMetricsFile);
         detailsFile.write(outputDetailMetricsFile);
 
+
+        if (!anyNonzeroLod){
```
Comment (pshapiro4broad) [inline src/main/java/picard/fingerprint/CheckFingerprint.java]:
```suggestion
        if (!anyNonzeroLod) {
```

File: src/main/java/picard/fingerprint/CheckFingerprint.java [hunk 2]
```diff
@@ -365,11 +366,22 @@ protected int doWork() {
 
             summaryFile.addMetric(metrics);
             log.info("Read Group: " + metrics.READ_GROUP + " / " + observedSampleAlias + " vs. " + metrics.SAMPLE + ": LOD = " + metrics.LOD_EXPECTED_SAMPLE);
+            if (metrics.LOD_EXPECTED_SAMPLE!=0) {
+                anyNonzeroLod |= true;
```
[the remark this answers is not in the document]
Comment (yfarjoun) [reply src/main/java/picard/fingerprint/CheckFingerprint.java]: "...and return 1" in what case? I am erroring out when I fail to find a non-zero LOD for _all_ the FPs in the loop....so I need to wait until I'm out of the loop

File: src/main/java/picard/fingerprint/CheckFingerprint.java [hunk 3]
```diff
@@ -366,16 +368,15 @@ protected int doWork() {
 
             summaryFile.addMetric(metrics);
             log.info("Read Group: " + metrics.READ_GROUP + " / " + observedSampleAlias + " vs. " + metrics.SAMPLE + ": LOD = " + metrics.LOD_EXPECTED_SAMPLE);
-            if (metrics.LOD_EXPECTED_SAMPLE!=0) {
-                anyNonzeroLod |= true;
+            if (metrics.LOD_EXPECTED_SAMPLE != 0) {
```
Comment (pshapiro4broad) [inline src/main/java/picard/fingerprint/CheckFingerprint.java]: you could write
```
anyNonzeroLod |= metrics.LOD_EXPECTED_SAMPLE != 0;
```
(also, I think the proper spelling is `anyNonZeroLod` as non-zero is a hyphenated word)

File: src/main/java/picard/fingerprint/CheckFingerprint.java [hunk 4]
```diff
@@ -224,7 +224,9 @@ protected int doWork() {
             outputDetailMetricsFile = DETAIL_OUTPUT;
             outputSummaryMetricsFile = SUMMARY_OUTPUT;
         } else {
-            if (!OUTPUT.endsWith(".")) OUTPUT = OUTPUT + ".";
+            if (!OUTPUT.endsWith(".")) {
+                OUTPUT = OUTPUT + ".";
```
Comment (pshapiro4broad) [inline src/main/java/picard/fingerprint/CheckFingerprint.java]:
```suggestion
                OUTPUT += ".";
```

File: src/main/java/picard/sam/markduplicates/MarkDuplicates.java [hunk 5]
```diff
@@ -268,38 +268,38 @@ protected int doWork() {
         // Key: previous PG ID on a SAM Record (or null).  Value: New PG ID to replace it.
         final Map<String, String> chainedPgIds = getChainedPgIds(outputHeader);
 
-        final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(outputHeader,
+        try(SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(outputHeader,
```
Comment (pshapiro4broad) [inline src/main/java/picard/sam/markduplicates/MarkDuplicates.java]:
```suggestion
        try (SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(outputHeader,
```

**Pull request merged**